File: Design Patterns/Command Pattern/Java/com.faculdadevincit.designpatterns.commandpattern/src/LampadaLigarCommand.java
public class LampadaLigarCommand implements ICommand {

    private Lampada lampada;

    public LampadaLigarCommand(Lampada lampada) {
        this.lampada = lampada;
    }

    @Override
    public void executar() {
        lampada.ligar();
    }

    @Override
    public void desfazer() {
        lampada.desligar();
    }

}

File: Design Patterns/Command Pattern/CSharp/FaculdadeVincit.DesignPatterns.CommandPattern/LampadaLigarCommand.cs
namespace FaculdadeVincit.DesignPatterns.CommandPattern;

public class LampadaLigarCommand : ICommand
{
    public Lampada lampada { get; private set; }

    public LampadaLigarCommand(Lampada lampada)
    {
        this.lampada = lampada;
    }

    public void Executar() => lampada.ligar();

    public void Desfazer() => lampada.desligar();
}
